msm/src/serialization/witness.rs

```diff
@@ -138,14 +138,18 @@ pub fn deserialize_field_element<Fp: Field, Env: InterpreterEnv<Fp>>(
     assert!(limbs[1] < 2u128.pow(88));
     assert!(limbs[2] < 2u128.pow(79));
 
-    env.copy(&input_limb0, kimchi_limbs0);
-    env.copy(&input_limb1, kimchi_limbs1);
+    let limb0_var = env.copy(&input_limb0, kimchi_limbs0);
```

Comment:
I am writing this here because I cannot write comments in non-diff lines.

I wonder why the input becomes constant variables. I would have imagined that whatever input to the msm you have, this will be included in a column of the gate. Constants are normally used for "real" constants like fixed numbers in equations, coefficients that are shared across all constraints, etc.

Reply:
`constant` is not the most appropriate function name I guess. It is to build a variable from a constant.

Reply:
But shouldn't you have that variable in your witness as a column already? I mean, I get it if you are in the witness code and just want to create the witness columns from witness data. But here it felt like this was constraints code, so the witness should already be created somewhere else, no?

Reply:
I consider the input of the witness builder (`deserialize_field_element` in this case) being “circuit agnostic”. The function must populate the appropriate “input” columns. The first 3 columns will be populated with the inputs. We might need to change this design later. I don't know yet how we will gather all circuits.